Creation of Landtrendr.AGB.Model [Work In Progress]

```diff
@@ -0,0 +1,203 @@
+    ## saved as HDF5 file format 
+      #if we export CSV but didn't provide any path
+      if(as.logical(export_csv) && is.null(outdir)){
+        PEcAn.logger::logger.info("If you want to export CSV file, please ensure input the outdir!")
+        return(0)
+      }
+    outdir = tempdir()
+```
+
+```{r}
+
+```
+
+
+```{r}
+
+ # Using data from GEDI4R package dataset (NASA website)
+  ## Create bounding box (search window) using varname 'country'
+    country_bound <- sf::st_as_sf(raster::getData('GADM', country = country, level = 1))
+         ## raster::getData; get geographic data for anywhere in the world; GADM (database) needs "country ="and level of subdivision 
+    #extract extent
+    e <- raster::extent(country_bound)
+    ## extent creates a matrix 
+    ul_lat <- e@ymax
+    lr_lat <- e@ymin
+    ul_lon <- e@xmin
+    lr_lon <- e@xmax
+    
+    #Get the list of all files available for the study area in the period selected,
+    #using just_path = T
+    file_path <- l4_download(
+      ul_lat,
+      lr_lat,
+      ul_lon,
+      lr_lon,
+      outdir = outdir,
+      from = start_date,
+      to = end_date,
+      just_path = T
+    )
+    
+    #download the first 4 files
+    file_download <- l4_download(
+      ul_lat,
+      lr_lat,
+      ul_lon,
+      lr_lon,
+      ncore = parallel::detectCores()-1,
+      outdir = outdir,
+      from = start_date,
+      to = end_date,
+      just_path = F,
+      subset=1:4
+    )
+    
+      l4_zip <- system.file("extdata",
+                          focus_sites,
+                          package="GEDI4R")
+    
+    #Unzipping GEDI level4A data
+    l4 <- lapply(l4_zip,unzip,exdir = outdir)
+    
+     ## Add columns not previously shown in gediL4 path data set
+    col <-
+      c(
+        "lat_lowestmode",
+        "lon_lowestmode",
+        "delta_time",
+        "agbd_pi_lower",
+        "agbd_pi_upper",
+        "agbd"
+      )
+    
+     gediL4_output <- l4_getmulti(l4,add_col = col, source= T)
+```
+```{r}
+# If using predownloaded dataset (saved as HDF5 file format) - (!) Could be automated to ask or check computer for file paths 
+    site_info = list.files(path =".", pattern = "h5", all.files = T, full.names = T)
+    View(site_info)
+    if(length(site_info) != 0){
+       dataname <- l4_getmulti(site_info[[1]],just_colnames = T)
+      gediL4_path <- l4_getmulti(site_info, catch = T, merge=T)
+      ## Dataset with predownloaded data
+        gediL4_pre <- l4_getmulti(site_info,just_colnames = F, add_col = col, source=T, catch = T)
+        rbind(gediL4_output, add_col = gediL4_pre, use.names = FALSE, fill = T) 
+    }
+   
+  
+
+```
+
+```{r}
+ ## Find and download GEDI Data within your study area:
+    ## site_info (list as data.table) tells the model the data files' site_id and its site_info 
+      #list all dataset in h5 file
+ 
+  ## Clipping Dataset
+  l4_data <- l4_getmulti(l4)
+  #clip using vector of coordinates
+  b_box <- c(-50,35,52,37)
```

Comment:
This seems like something that's leftover from the tutorial you were building from, not something that should be part of the function.  More generally, does our function need to clip to a bounding box in order to extract data for a site, or do we end up doing that in another way? If the latter this can be skipped. If the former, then we probably need a loop where we clip each site +/- whatever we decide the search radius is for an individual site.